tests/scripts/basic-build-test.sh

```diff
@@ -185,6 +230,11 @@ echo "Total exec'd tests : $TOTAL_EXED"
 echo "Total avail tests  : $TOTAL_AVAIL"
 echo
 
+# Set exitcode if any tests failed
+if [ $(echo "$TOTAL_FAIL > 0" | bc) = "1" ]; then
```

Comment:
I think that `$TOTAL_FAIL` should actually be an integer, so I suggest to use the regular arguments to the test command to avoid further use of `bc` as you said its problematic. For example:
```
if [ $TOTAL_FAIL -gt 0 ]; then
```